pkg/components/data_node.go

```diff
@@ -29,7 +30,10 @@ func NewDataNode(
 		APIProxy:       ytsaurus.APIProxy(),
 		ComponentLabel: cfgen.FormatComponentStringWithDefault(consts.YTComponentLabelDataNode, spec.Name),
 		ComponentName:  cfgen.FormatComponentStringWithDefault("DataNode", spec.Name),
-		MonitoringPort: consts.DataNodeMonitoringPort,
```

Comment:
You could leave monitoring port on labeler but fetch it from spec:

`MonitoringPort: ptr.Int32Deref(spec.InstanceSpec.MonitoringPort, consts.DataNodeMonitoringPort)`

Reply:
I intentionally dropped it from here since I don't see any reason to be inside. Monitoring/api port seems to have neither relation to the 'labeller' (how the struct is named) logic

Reply:
Yeah, maybe it needs better name or refactoring,
but this pr cloud be much smaller with leaving port here.